File: src/frontend/src/entities/Topic.ts
import { apiClient } from '~/shared/apiClient'

export interface Topic {
  code: string
  title: string
  lessons: string[]
  traits: string[]
  description: string
}

export const TopicsApi = {
  async getAll() {
    const { data } = await apiClient<Topic[]>('/Topics/GetTopics')

    return data
  },
}

File: src/backend/backend.Core/JsonModels/Topic.cs
namespace backend.Core.JsonModels;

public class JsonTopic
{
    public string Code { get; set; }
    public string Title { get; set; }
    public List<string> Lessons { get; set; }
    public List<string> Traits { get; set; }
    public string Description { get; set; }
}